consul/consul_test.go

```diff
@@ -1001,3 +1001,47 @@ func TestMarathonTaskToConsulServiceMapping_NotResolvableTaskHost(t *testing.T)
 	// then
 	assert.Error(t, err)
 }
+
+func Test_substituteEnvironment(t *testing.T) {
+	type args struct {
+		s    string
+		task apps.Task
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "without port and host",
+			args: args{
+				s: "true",
+				task: apps.Task{
+					ID:    "someTask",
+					Host:  "127.0.0.6",
+					Ports: []int{8090, 8443},
+				},
+			},
+			want: "true",
+		},
+		{
+			name: "with ports and host",
+			args: args{
+				s: "nc $HOST $PORT0 && nc $HOST $PORT1",
```

Comment:
Yes that can be a problem... but solving this can be very difficult. I found this package https://godoc.org/github.com/mvdan/sh/syntax to parse and "understand" shell but API is not simple.